### PR TITLE
Feature: ability to skip adding source directories if the directories do not exist

### PR DESCRIPTION
**Pull request by viktor-yengovatov**
By this feature non exiting source directories like `kotlintestsrc` are excluded from source directories.

The feature is disabled by default.
 
Signed-off-by: Your Real Name your.email@email.com

### CODE REVIEW

**Comment by mlytvyn** [comment]
Hello @viktor-yengovatov , thank you very much for your efforts and contribution!!!
It would be great if you could also update `CHANGELOG.md` file with something like:
```
## [2023.2.4]

### Features
- Added possibility to skip non-existing source directories during project import [#511](https://github.com/epam/sap-commerce-intellij-idea-plugin/pull/511) 

```

File: src/com/intellij/idea/plugin/hybris/project/configurators/impl/RegularContentRootConfigurator.java [hunk 1]
```diff
@@ -328,6 +330,10 @@ private static <P extends JpsElement> void addSourceFolderIfNotIgnored(
         @NotNull final List<File> dirsToIgnore
     ) {
         if (dirsToIgnore.stream().noneMatch(it -> FileUtil.isAncestor(it, testSrcDir, false))) {
+            final boolean ignoreEmpty = HybrisApplicationSettingsComponent.getInstance().getState().getIgnoreNonExistingSourceDirectories();
```
Comment (mlytvyn) [inline src/com/intellij/idea/plugin/hybris/project/configurators/impl/RegularContentRootConfigurator.java:333]: @viktor-yengovatov , what do you think if we can refactor this part slightly and pass state of the `HybrisApplicationSettingsComponent` to this class -> methods from the `ImportProjectProgressModalWindow`
by that we will get appSettings only once during import of all java modules?

variable can be declared in the `public synchronized void run(@NotNull final ProgressIndicator indicator) {` of the `ImportProjectProgressModalWindow` and then passed to `createJavaModule` then to `configuratorFactory.getContentRootConfigurator(moduleDescriptor).configure` and so on

Comment (viktor-yengovatov) [reply src/com/intellij/idea/plugin/hybris/project/configurators/impl/RegularContentRootConfigurator.java:333]: Hello @mlytvyn 
I have created separate pull request for the refactoring: #512 .
I hope it fits.

Comment (mlytvyn) [reply src/com/intellij/idea/plugin/hybris/project/configurators/impl/RegularContentRootConfigurator.java:333]: sure, just take into account different target branch, I'll also change target branch for your PR #512

**Pull request merged**